packages/ui/src/components/chip/chip.native.tsx

```diff
@@ -0,0 +1,28 @@
+import { createBox } from '@shopify/restyle';
+
+import { Text } from '../../../native';
+import { type Theme } from '../../theme-native';
+
+interface ChipProps {
+  label: string | number;
+}
+
+export const Chip = createBox<Theme, ChipProps>(({ label, ...rest }) => (
+  <Text variant="label02" width="5" height="4" {...rest}>
+    {label}
+  </Text>
+));
+
+//  FIXME double check padding and gap with design
+Chip.defaultProps = {
+  backgroundColor: 'ink.component-background-default',
+  borderRadius: 'xs',
+  paddingLeft: '1',
+  paddingRight: '1',
+  paddingTop: '0', // says 2px in figma but no token for that
+  paddingBottom: '0', // says 2px in figma but no token for that
+  flexDirection: 'column',
+  justifyContent: 'center',
+  alignItems: 'center',
+  gap: '3', // 12px - says 10px in figma but no token for that
+};
```

Comment:
https://react.dev/blog/2024/04/25/react-19-upgrade-guide#removed-deprecated-react-apis

Believe we shouldn't be using `defaultProps`. But we can just add the props to the compoent, and allow them to be overridden instead?

Reply:
OK let me check that. I am doing it this way as it's easer to extend `Chip` to accept more props if you use `createBox<Theme, ChipProps>`.

I did a similar thing in `hstack.native.tsx`. I'm not sure how to do our usual `interface Chip extends BoxProps {`  with the native `Box` component but I can check it out

Reply:
@kyranjamie I just made a [different task](https://github.com/leather-io/issues/issues/324) for refactoring this as I don't want to get deep into it right now. 

I'll change it shortly once I come up with a better way